src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleBinaryOperator.cs

```diff
@@ -148,8 +170,17 @@ private BoundExpression ReplaceTerminalElementsWithTemps(BoundExpression expr, T
             var isRightNullable = right.Kind != BoundKind.TupleLiteral && right.Type.IsNullableType();
             if (isRightNullable)
             {
-                rightHasValue = MakeNullableHasValue(right.Syntax, right); // no need for local for right.HasValue since used once
-                rightValue = MakeValueOrDefaultTemp(right, temps, innerEffects);
+                BoundExpression value = NullableAlwaysHasValue(right);
+                if (value is null)
+                {
+                    rightHasValue = MakeNullableHasValue(right.Syntax, right); // no need for local for right.HasValue since used once
+                    rightValue = MakeValueOrDefaultTemp(right, temps, innerEffects);
+                }
+                else
+                {
+                    rightHasValue = MakeBooleanConstant(right.Syntax, true);
+                    rightValue = value;
```

Comment:
>rightValue = value; [](start = 20, length = 19)

```isRightNullable = false;``` ?

Reply:
It is used on line 201 below if (!isLeftNullable && !isRightNullable)


---
In reply to: [176278184](https://github.com/dotnet/roslyn/pull/25644#discussion_r176278184) [](ancestors = 176278184)

Reply:
Ah, thanks

---
In reply to: [176501791](https://github.com/dotnet/roslyn/pull/25644#discussion_r176501791) [](ancestors = 176501791,176278184)